### ISSUE TITLE
request.json() / request.text() doesn't resolve when "Content-Type" header is "application/json"

### ISSUE THREAD

**Issue by drichardcarl** [remix-run/remix#9857]
### Reproduction

1. Create a route that export action only
```ts
// app/routes/test.tsx
export async function action({
  request,
}: ActionFunctionArgs) {
  const body = await request.json(); // or await request.text()
 // This never gets here
  return redirect(`/`);
}
```

2. Send POST request to the endpoint `http://localhost:5173/test` using fetcher, fetch, Postman, curl, etc. as long as you send with "Content-Type": "application/json" header.
```sh
> curl -X POST http://localhost:5173/test -H "Content-Type: application/json" -d '{"key": "value"}'
```

3. `await request.json()` resolves just fine if "Content-Type" header is set to other value, e.g. "text/plain"

### System Info

```shell
System:
    OS: Linux 6.9 Manjaro Linux
    CPU: (8) x64 AMD Ryzen 7 4700U with Radeon Graphics
    Memory: 12.48 GB / 30.72 GB
    Container: Yes
    Shell: 5.2.26 - /bin/bash
  Binaries:
    Node: 22.2.0 - ~/.nvm/versions/node/v22.2.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.2.0/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v22.2.0/bin/npm
  npmPackages:
    @remix-run/dev: ^2.9.1 => 2.9.1 
    @remix-run/express: ^2.10.0 => 2.10.0 
    @remix-run/node: ^2.9.1 => 2.9.1 
    @remix-run/react: ^2.9.1 => 2.9.1 
    @remix-run/v1-route-convention: ^0.1.4 => 0.1.4 
    vite: ^5.1.0 => 5.2.11
```

### Used Package Manager

npm

### Expected Behavior

`await request.json()` should resolve as expected

### Actual Behavior

`await request.json()` never gets resolved

**Comment by brophdawg11** [comment]
Can you provide a minimal reproduction?  This is working fine for me both from a `fetcher` and the `curl` command you included.

**Comment by brophdawg11** [comment]
I also noticed your Remix express dependency does not match the other remix versions.  Please try with the latest release and ensure your Remix express dependency matches the others (dev, node, react)

**Comment by drichardcarl** [comment]
Hi @brophdawg11 , sorry for taking too long to respond. I've tried using the same version for all @remix-run/* packages to no avail.

```sh
  System:
    OS: Linux 6.9 Manjaro Linux
    CPU: (8) x64 AMD Ryzen 7 4700U with Radeon Graphics
    Memory: 7.90 GB / 30.72 GB
    Container: Yes
    Shell: 5.2.26 - /bin/bash
  Binaries:
    Node: 22.2.0 - ~/.nvm/versions/node/v22.2.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.2.0/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v22.2.0/bin/npm
  npmPackages:
    @remix-run/dev: ^2.9.1 => 2.9.1 
    @remix-run/express: ^2.9.1 => 2.9.1 
    @remix-run/node: ^2.9.1 => 2.9.1 
    @remix-run/react: ^2.9.1 => 2.9.1 
    @remix-run/v1-route-convention: ^0.1.4 => 0.1.4 
    vite: ^5.1.0 => 5.2.11
```

**Comment by drichardcarl** [comment]
> Hi @brophdawg11 , sorry for taking too long to respond. I've tried using the same version for all @remix-run/* packages (dev, node, react, express) to no avail.
> 
> ```shell
>   System:
>     OS: Linux 6.9 Manjaro Linux
>     CPU: (8) x64 AMD Ryzen 7 4700U with Radeon Graphics
>     Memory: 7.90 GB / 30.72 GB
>     Container: Yes
>     Shell: 5.2.26 - /bin/bash
>   Binaries:
>     Node: 22.2.0 - ~/.nvm/versions/node/v22.2.0/bin/node
>     Yarn: 1.22.22 - ~/.nvm/versions/node/v22.2.0/bin/yarn
>     npm: 10.7.0 - ~/.nvm/versions/node/v22.2.0/bin/npm
>   npmPackages:
>     @remix-run/dev: ^2.9.1 => 2.9.1 
>     @remix-run/express: ^2.9.1 => 2.9.1 
>     @remix-run/node: ^2.9.1 => 2.9.1 
>     @remix-run/react: ^2.9.1 => 2.9.1 
>     @remix-run/v1-route-convention: ^0.1.4 => 0.1.4 
>     vite: ^5.1.0 => 5.2.11
> ```

I also tried updating to latest version (dev, express, node, react) but same issue.

```sh
  System:
    OS: Linux 6.9 Manjaro Linux
    CPU: (8) x64 AMD Ryzen 7 4700U with Radeon Graphics
    Memory: 7.70 GB / 30.72 GB
    Container: Yes
    Shell: 5.2.26 - /bin/bash
  Binaries:
    Node: 22.2.0 - ~/.nvm/versions/node/v22.2.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.2.0/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v22.2.0/bin/npm
  npmPackages:
    @remix-run/dev: ^2.11.2 => 2.11.2 
    @remix-run/express: ^2.11.2 => 2.11.2 
    @remix-run/node: ^2.11.2 => 2.11.2 
    @remix-run/react: ^2.11.2 => 2.11.2 
    @remix-run/v1-route-convention: ^0.1.4 => 0.1.4 
    vite: ^5.1.0 => 5.2.11
```

**Comment by brophdawg11** [comment]
Can you provide a minimal reproduction showing the issue?  I was unable to reproduce locally based on what you have provided.

**Comment by drichardcarl** [comment]
Interesting enough. I've spin-up a minimal remix app using `npx create-remix@latest` and I can't reproduce the issue as well. I did the same thing to reproduce on a remix app I'm working on (which I encountered this issue originally) and the issue only exists in that app!

```sh
  System:
    OS: Linux 6.9 Manjaro Linux
    CPU: (8) x64 AMD Ryzen 7 4700U with Radeon Graphics
    Memory: 21.48 GB / 30.72 GB
    Container: Yes
    Shell: 5.2.26 - /bin/bash
  Binaries:
    Node: 22.2.0 - ~/.nvm/versions/node/v22.2.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.2.0/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v22.2.0/bin/npm
  npmPackages:
    @remix-run/dev: ^2.11.2 => 2.11.2 
    @remix-run/node: ^2.11.2 => 2.11.2 
    @remix-run/react: ^2.11.2 => 2.11.2 
    @remix-run/serve: ^2.11.2 => 2.11.2 
    vite: ^5.1.0 => 5.4.2
```

**Comment by brophdawg11** [comment]
ok - it seems it's something related to that specific app setup.  I'm going to close this since it doesn't appear to be a Remix issue - but happy to re-open if you are able to find a minimal reproduction.

I would try trimming portions/customizations of the app one by one and see if that helps you narrow down the root cause.

**Comment by drichardcarl** [comment]
So, after couple of months I decided to revisit this.

After a couple of resets on my commit history to try and check which commit affected this, I finally managed to find the culprit: it was the custom express server I did.

Technically, it's not the custom express server (https://remix.run/docs/en/main/start/quickstart#development-workflow) but something I did in the file.

The line commented-out is the one causing this issue.
![image](https://github.com/user-attachments/assets/bd14ef33-5085-4160-afd3-79b26efad793)

So that's it.
I'd like to share this for awareness.

Thanks @brophdawg11 for the help.
